x/dclauth/keeper/msg_server_reject_add_account.go

```diff
@@ -0,0 +1,94 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/dclauth/types"
+)
+
+func (k msgServer) RejectAddAccount(
+	goCtx context.Context, msg *types.MsgRejectAddAccount,
+) (*types.MsgRejectAddAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Signer: (%s)", err)
+	}
+
+	// check if sendor has enough rights to create a validator node
+	if !k.HasRole(ctx, signerAddr, types.Trustee) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized,
+			"MsgApproveAddAccount transaction should be signed by an account with the %s role",
+			types.Trustee,
+		)
+	}
+
+	accAddr, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+
+	// check if pending account exists
+	if !k.IsPendingAccountPresent(ctx, accAddr) {
+		return nil, types.ErrPendingAccountDoesNotExist(msg.Address)
+	}
+
+	// get pending account
+	pendAcc, _ := k.GetPendingAccount(ctx, accAddr)
+
+	// check if pending account already has reject approval from signer
+	if pendAcc.HasRejactApprovalFrom(signerAddr) {
```

Comment:
This block seems duplicated. Think about refactoring it to a single function

Reply:
Decided not to change! Due to the fact that these errors are shown in more detail, where these errors were reported